### PR TITLE
aarch64: extend matmul heuristic checks to all neoverse platforms

### PR DESCRIPTION
**Pull request by snadampal**
for aarch64 neoverse platforms there are two gemm backends available
for matmul operator on PyTorch: (1) Arm Compute Library and (2) OpenBLAS.
While Arm Compute Library provides better performance over OpenBLAS,
it has overhead for the kernel launch time, and hence we use OpenBLAS
for smaller tensor compute. The heuristic was originally implemented for 
neoverse_v1. This commit extends the heuristic to other neoverse platforms

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10

### CODE REVIEW

**Comment by snadampal** [comment]
Hi @malfet , could you please review and approve if it looks good.

**Comment by malfet** [comment]
@snadampal overall looks good, but can you please provide a short description on why this change is needed?

File: aten/src/ATen/native/LinearAlgebra.cpp
Comment (malfet) [inline aten/src/ATen/native/LinearAlgebra.cpp:1364]: Can you please elaborate why this heuristic should be applied to neonvese_v2 but not to neonverse_n2?
Also, considering the code is almost identical between `get_mkldnn_matmul_min_dim` and `get_mkldnn_matmul_min_size` do you mind refactoring this into `is_arm_neonverse()` helper routine

**Review by malfet — CHANGES_REQUESTED** [review]
@snadampal please ping me on PR when this change will apply to all neonverse arches and condition is refactored into a common function

**Review by malfet — APPROVED** [review]
Thank you for the updates, looks good to me

**Comment by snadampal** [comment]
@malfet , thanks for initiating the merge. I was waiting for the CI runs to complete to ping you.

**Pull request closed without merging**
